### DISCUSSION TITLE
How to switch to another branch

### DISCUSSION THREAD

**Discussion by tomyang03** [community/community#22552] [New to GitHub]
<p>I am using git checkout master, but when i do git branch -a , it is still pointing to branch2.</p>
<p>PS C:\Node\projects\n-5-10-workflow-test&gt; git branch -a<br>
* branch2<br>
master<br>
remotes/heroku/master<br>
remotes/origin/branch2<br>
remotes/origin/master<br>
PS C:\Node\projects\n-5-10-workflow-test&gt; git checkout master<br>
error: Your local changes to the following files would be overwritten by checkout:<br>
server.js<br>
Please commit your changes or stash them before you switch branches.<br>
Aborting<br>
PS C:\Node\projects\n-5-10-workflow-test&gt; git add .<br>
warning: LF will be replaced by CRLF in server.js.<br>
The file will have its original line endings in your working directory.<br>
PS C:\Node\projects\n-5-10-workflow-test&gt; git commit -m “10th commit”<br>
[branch2 b5c7811] 10th commit<br>
1 file changed, 1 insertion(+), 2 deletions(-)<br>
PS C:\Node\projects\n-5-10-workflow-test&gt; git push origin master<br>
Everything up-to-date<br>
PS C:\Node\projects\n-5-10-workflow-test&gt; git commit -m “11th commit”<br>
On branch branch2<br>
nothing to commit, working tree clean<br>
PS C:\Node\projects\n-5-10-workflow-test&gt;</p>

**Comment by aster94** [comment]
<p>try this <a href="https://git-scm.com/docs/git-checkout" rel="nofollow noopener">https://git-scm.com/docs/git-checkout</a></p>

**Comment by Wabri** [comment, marked as answer]
<p>If you are in branch2 you need to push to branch2 and not in master, with git push origin master you push your local master to origin. If the changes are made in branch2 you had to push to branch2:</p>
<pre><code>git push origin branch2
</code></pre>
<p>then you can switch to master with:</p>
<pre><code>git checkout master
</code></pre>
<p>In order:</p>
<ol>
<li>
<p>The first command you run is:</p>
<p>git branch -a</p>
</li>
</ol>
<p>That show the list both <strong>remote-tracking</strong> branches and <strong>local</strong> branches.</p>
<ol start="2">
<li>
<p>You tried switch to local master using checkout, but there are changes in the branch2 so the checkout command was <strong>aborted</strong>.</p>
</li>
<li>
<p>You added and committed the changes in <strong>branch2</strong> using:</p>
<p>git add .<br>
git commit -m “10th commit”</p>
</li>
<li>
<p>Now the <strong>error</strong> , you pushed to origin master the changes, but there are no changes made on local master so the command:</p>
<p>git push origin master</p>
</li>
</ol>
<p>Gived the output:</p>
<pre><code>Everything up-to-date
</code></pre>
<p>And this is right, because your precedent commit was made on branch2. You need to push branch2 to origin, with this:</p>
<pre><code>git push origin branch2
</code></pre>
<ol start="5">
<li>Last command didnt have sense to run, because there are no staged files.</li>
</ol>
<p>After push to origin branch2 you can run:</p>
<pre><code>git checkout master
</code></pre>
<p>and it will be succesfull.</p>
<p>Hope to solve your problems.</p>
<p>-Gabriele-</p>

**Comment by mpboom** [comment]
<p>Exactly what <a class="mention" href="/wabri">@wabri</a> says; your checkout gets aborted because Git could then possible destroy files. See this line:</p>
<pre><code>error: Your local changes to the following files would be overwritten by checkout:
server.js
</code></pre>
